### ISSUE TITLE
Broken markdown link referencing "media type range"

### ISSUE THREAD

**Issue by saibotsivad** [OAI/OpenAPI-Specification#2796]
On the OpenAPI specs page for [Request Body Object](https://spec.openapis.org/oas/latest.html#request-body-object) what is loaded in my browser looks like this (note the highlighted text):

<img width="812" alt="image" src="https://user-images.githubusercontent.com/91254/142489647-c065dd57-262f-4960-a861-2227344eb399.png">

I believe that comes from [/versions/3.1.0.md](https://raw.githubusercontent.com/OAI/OpenAPI-Specification/a114313cf1558877bce771bdad7d3b095a4f48f5/versions/3.1.0.md) but the Markdown looks correct to me:

```markdown
The key is a media type or [media type range](https://tools.ietf.org/html/rfc7231#appendix-D) and the value describes it.
```

I tried copy+pasting the surrounding section over to the [markdown-it online demo](https://markdown-it.github.io/) to see if maybe the table syntax was buggy, but it renders correctly there.

Is there a possibility that the published version (at `spec.openapis.org/oas/latest.html`) is not actually generated from the latest markdown file?

I'm not sure what else to do to be helpful, sorry!

**Comment by webron** [comment]
Thanks for bringing this up. My guess is that this is a rendering issue related to anchored URL. We'll need to take a look at the generation process in order to solve it.

**Comment by saibotsivad** [comment]
Yeah, that must be it. I thought it was only that one place, but now I see it all over the docs, e.g. in the Operation Object `requestBody` text:

<img width="804" alt="image" src="https://user-images.githubusercontent.com/91254/142686867-a024c7c7-6ae9-4440-a4c7-a145e1fcf868.png">

which has this markdown text:

```markdown
... (such as [GET](https://tools.ietf.org/html/rfc7231#section-4.3.1), ...
```

**Comment by saibotsivad** [comment]
I believe the problem is to do with this regex-cleanup stuff here: https://github.com/OAI/OpenAPI-Specification/blob/main/scripts/md2html/md2html.js#L262

@webron is this something you'd appreciate a pull request for? If so, from/into what branch?

**Comment by webron** [comment]
We always welcome PRs, and in this case against `main`. Not sure how to test it though, but will be happy to give it a shot.

**Comment by karenetheridge** [comment]
I also see something similar on the github site at https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#fixed-fields - in "webhooks", the text is `Map[string, Path Item Object | Reference Object] ]`

**Comment by saibotsivad** [comment]
@karenetheridge that looks unrelated, the actual markdown has the extra trailing `]`:

```markdown
<a name="oasWebhooks"></a>webhooks | Map[`string`, [Path Item Object](#pathItemObject) \| [Reference Object](#referenceObject)] ] | The incoming webhooks...
```

**Comment by saibotsivad** [comment]
I created pull request #2798 to `main` with the fix, and a description of the problem+solution.

**Comment by karenetheridge** [comment]
I created https://github.com/OAI/OpenAPI-Specification/pull/2831 for the unrelated markdown error in the webhooks section.

**Comment by MikeRalphson** [comment]
Closing as the cause was fixed in #2798 - thanks all!
